File: cpp/1212.cpp
#include <iostream>
#include <string>

using namespace std;

int main()
{
    string a = "", b = "";

    while (true) {
        cin >> a >> b;

        if (a == "0" && b == "0") {
            break;
        }

        int len = a.length(), len_b = b.length();

        while (len < len_b) {
            a = "0" + a;
            len++;
        }

        while (len_b < len) {
            b = "0" + b;
            len_b++;
        }

        len--;
        long long int carry = 0, carry_operations = 0, current_sum;

        while (true) {
            current_sum = (int)(a[len]) + (int)(b[len]) - 96;

            if (carry > 0) {
                current_sum += carry;
                carry = 0;
            }

            if (current_sum > 9) {
                carry++;
                carry_operations++;
            }

            a.pop_back();
            b.pop_back();
            len--;

            if (len == -1) {
                break;
            }
        }

        if (carry_operations > 1) {
            cout << carry_operations << " carry operations.\n";
        } else if (carry_operations == 1) {
            cout << "1 carry operation.\n";
        } else {
            cout << "No carry operation.\n";
        }
    }

    return 0;
}

File: py/1212.py
while True:
    n = input().split()
    
    if n == ['0', '0']:
        break
    
    while len(n[0]) < len(n[1]):
        n[0] = '0' + n[0]
        
    while len(n[1]) < len(n[0]):
        n[1] = '0' + n[1]
    
    n0 = [i for i in n[0]]
    n1 = [i for i in n[1]]
    
    carry = 0
    carryOperations = 0
    
    while True:
        currentSum = int(n0[-1]) + int(n1[-1])
        
        if carry > 0:
            currentSum += carry
            carry = 0
        
        if currentSum > 9:
            carry += 1
            carryOperations += 1
            
        n0.pop(-1)
        n1.pop(-1)
        
        if len(n0) == 0 or len(n1) == 0:
            break
        
    if carryOperations > 1:
        print(f'{carryOperations} carry operations.')
        
    elif carryOperations == 1:
        print('1 carry operation.')
        
    else:
        print('No carry operation.')